src/Akka.Persistence.Sql.Linq2Db/Journal/DAO/BaseByteArrayJournalDao.cs

```diff
@@ -374,31 +374,75 @@ await db.GetTable<JournalMetaData>()
                     SequenceNumber = md.sequenceNumber,
                     PersistenceId = md.persistenceId
                 };
-        private IQueryable<long> MaxSeqNumberForPersistenceIdQuery(
+        private IQueryable<long?> MaxSeqNumberForPersistenceIdQuery(
             DataConnection db, string persistenceId, long minSequenceNumber = 0)
         {
-            
-            
-
-            var queryable = db.GetTable<JournalRow>()
-                .Where(r => r.persistenceId == persistenceId).Select(rowDataSelector);
             if (minSequenceNumber != 0)
             {
-                queryable = queryable.Where(r =>
-                    r.SequenceNumber > minSequenceNumber);
+                if ((_journalConfig.DaoConfig.SqlCommonCompatibilityMode))
+                {
+                    return MaxSeqForPersistenceIdQueryableCompatibilityModeWithMinId(db, persistenceId, minSequenceNumber);
+                }
+                else
+                {
+                    return MaxSeqForPersistenceIdQueryableNativeModeMinId(db, persistenceId, minSequenceNumber);
+                }
             }
-
-            if (_journalConfig.DaoConfig.SqlCommonCompatibilityMode)
+            else
             {
-                var nextQuery = db.GetTable<JournalMetaData>()
+                if (_journalConfig.DaoConfig.SqlCommonCompatibilityMode)
+                {
+                    return MaxSeqForPersistenceIdQueryableCompatibilityMode(db, persistenceId);
+                }
+
+                return MaxSeqForPersistenceIdQueryableNativeMode(db, persistenceId);
+            }
+        }
+
+        private static IQueryable<long?> MaxSeqForPersistenceIdQueryableNativeMode(DataConnection db,
+            string persistenceId)
+        {
+            return db.GetTable<JournalRow>()
+                .Where(r => r.persistenceId == persistenceId)
+                .Select(r => (long?)r.sequenceNumber);
```

Comment:
So, to make best SQL we have to do a `.MaxAsync()` in the main call (`.FirstOrDefault` results in `TAKE` semantics and additional param pass), so for this case we don't do any constraining.

This is the happy path, we will wind up with SQL like:

```sqlite
SELECT
	Max([r].[sequence_number])
FROM
	[journal] [r]
WHERE
	[r].[persistence_id] = @persistenceId
```

Reply:
LGTM